src/frontend/specification/EnsembleParameterTable.jl

```diff
@@ -2,50 +2,73 @@
 ### Types
 ############################################################################################
 
-mutable struct EnsembleParameterTable{C} <: AbstractParameterTable
-    tables::C
+struct EnsembleParameterTable <: AbstractParameterTable
+    tables::Dict{Symbol, ParameterTable}
+    params::Vector{Symbol}
 end
 
 ############################################################################################
 ### Constructors
 ############################################################################################
 
 # constuct an empty table
-function EnsembleParameterTable(::Nothing)
-    tables = Dict{Symbol, ParameterTable}()
-    return EnsembleParameterTable(tables)
+EnsembleParameterTable(::Nothing; params::Union{Nothing, Vector{Symbol}} = nothing) =
+    EnsembleParameterTable(
+        Dict{Symbol, ParameterTable}(),
+        isnothing(params) ? Symbol[] : copy(params),
+    )
+
+# dictionary of SEM specifications
+function EnsembleParameterTable(
+    spec_ensemble::AbstractDict{K, V};
+    params::Union{Nothing, Vector{Symbol}} = nothing,
+) where {K, V <: SemSpecification}
+    params = if isnothing(params)
+        # collect all SEM parameters in ensemble if not specified
+        # and apply the set to all partables
+        unique(mapreduce(SEM.params, vcat, values(spec_ensemble), init = Vector{Symbol}()))
+    else
+        copy(params)
+    end
+
+    # convert each model specification to ParameterTable
+    partables = Dict{Symbol, ParameterTable}(
+        Symbol(group) => convert(ParameterTable, spec; params) for
+        (group, spec) in pairs(spec_ensemble)
+    )
+    return EnsembleParameterTable(partables, params)
 end
 
```

Comment:
Maybe we want to have something like this:
```suggestion

EnsembleParameterTable(ps::Pair{K, V}...; params = nothing) where {K, V} = EnsembleParameterTable(Dict(ps...); params=params)

```
So users can just write
```
partable = EnsembleParameterTable(:Pasteur => specification_g1, :Grant_White => specification_g2))
```
instead of having to construct the Dict themselves every time:
```
partable = EnsembleParameterTable(
    Dict(:Pasteur => specification_g1, :Grant_White => specification_g2),
)
```

Reply:
I like your suggestion. For this (already extensive) PR I tried to limit changes to the public API.
Basically, I touched EnsParTable only for the internal cleanups and to make sure all subtypes of *SemSpecification* can provide *params* and observed/latent vars.
I suggest that we implement more user-visible changes once this PR is merged.

Reply:
Agreed.